tunder/lib/src/database/query.dart

```diff
@@ -37,14 +42,26 @@ class Query<T> {
     this.table = _inferTableNameVia(tableNameOrModelClass);
   }
 
+  Future<List<T>> get() async => _transformRows(await _operation.process(this));
+
+  Future<int> count() => CountOperation.forDatabase(DB.driver).process(this);
+
+  Future<int> insert(Map<String, dynamic> row) =>
```

Comment:
Taking a quick look I can't tell which method is the read operation: get(), all() or select(). The API may be more clear if the CRUD method names are create, read, update and delete. Nevertheless, after writing the code once the dev may know which method is the expected operation.

Reply:
In this case the method names follow the convention from SQL databases: insert, update, delete. Except for the select that really doesn't retrieve by calling it. The `get` and `all` method name is common for people coming from Laravel and rails. I don't think it's complicated in this case though. Interesting suggestion about the `read` method which throws an exception, but I don't think it is worth introducing a method that doesn't do anything.